pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -31,19 +32,64 @@
 class LoggerConnector:
     def __init__(self, trainer):
         self.trainer = trainer
-        self.callback_metrics = {}
-        self.evaluation_callback_metrics = {}
-        self.logged_metrics = {}
-        self.progress_bar_metrics = {}
+        self._callback_metrics = MetricsHolder()
```

Comment:
I very much agree with this protection, but in such case we probably need to keep API back-compatible:
- add the past as property with a warning whenever used
- include in a warning deprecation warning
- add tests that the separation warning is properly called

Reply:
It is still backward compatible as I moved them as property.

Reply:
yes, but do we deprecate it in the future?

Reply:
I don't understand why you think we should deprecate something.

Reply:
It basically just add property to LoggerConnector, it calls convert every time the user request the metrics and don't when we gather metrics internally as moved into private variable.